web/js/modules/product-picker/util.js

```diff
@@ -0,0 +1,81 @@
+import { cloneDeep as lodashCloneDeep } from 'lodash';
+import safeLocalStorage from '../../util/local-storage';
+
+const { RECENT_LAYERS } = safeLocalStorage.keys;
+const MAX_RECENT_LAYERS = 20;
+const DEFAULT_OBJ = {
+  geographic: [],
+  arctic: [],
+  antarctic: [],
+};
+
+export function getRecentLayers(layerConfig, proj) {
+  const byUse = (a, b) => {
+    if (a.count > b.count) return -1;
+    if (a.count < b.count) return 1;
+    if (a.dateAdded > b.dateAdded) return 1;
+    if (a.dateAdded < b.dateAdded) return -1;
+  };
+  const toLayerObj = ({ id }) => layerConfig[id];
+  const layers = JSON.parse(safeLocalStorage.getItem(RECENT_LAYERS));
+  return layers ? layers[proj].sort(byUse).map(toLayerObj) : [];
+}
+
+export function clearRecentLayers() {
+  safeLocalStorage.removeItem(RECENT_LAYERS);
+}
+
+/**
+ *
+ * @param {*} layer - the layer being added to the map
+ */
+export function clearSingleRecentLayer({ id: layerId, projections }) {
+  const recentLayersJson = safeLocalStorage.getItem(RECENT_LAYERS);
+  const recentLayers = JSON.parse(recentLayersJson) || lodashCloneDeep(DEFAULT_OBJ);
+  Object.keys(projections).forEach((proj) => {
+    const layers = recentLayers[proj];
+    recentLayers[proj] = layers.filter(({ id }) => id !== layerId);
+  });
+  safeLocalStorage.setItem(RECENT_LAYERS, JSON.stringify(recentLayers));
+}
+
+/**
+ *
+ * @param {*} layer - the layer being added to the map
+ */
+export function updateRecentLayers({ id: layerId, projections }) {
+  const recentLayersJson = safeLocalStorage.getItem(RECENT_LAYERS);
+  const recentLayers = JSON.parse(recentLayersJson) || lodashCloneDeep(DEFAULT_OBJ);
```

Comment:
```suggestion
  const recentLayers = (safeLocalStorage.enabled && JSON.parse(recentLayersJson)) || lodashCloneDeep(DEFAULT_OBJ);
```
This should prevent a nonbreaking JSON parse console error if local storage is disabled while adding a layer.

Reply:
Good catch!  I will probably go with an early return if localStorage is disabled since the rest of the function becomes pointless in those cases:

```
export function updateRecentLayers({ id: layerId, projections }) {
  if (!safeLocalStorage.enabled) {
    return;
  }
...
}
```